src/motion_pvt_sequence_generation/README.md

```diff
@@ -0,0 +1,250 @@
+# PVT Sequence Generation
+
+*By Jeff Homer, Firmware Team*
+
+This repository contains code to complement our article [Motion Planning with Position-Velocity-Time](https://www.zaber.com/articles/motion-planning-with-position-velocity-time). The cubic polynomial PVT algorithm
+described in the article is implemented in Python, along with helper functions to automatically
+generate missing parameters, plot the generated path and trajectory, and save the results
+to a CSV file that is compatible with [Zaber Launcher](https://software.zaber.com/zaber-launcher/download)'s PVT Viewer App.
+
+## Hardware Requirements
+This code can be used to generate PVT sequences that can be run on devices with [Zaber](https://www.zaber.com/) controllers.
+However, the underlying math and generated trajectories could be utilized on almost any motion controller with position-velocity-time capabilities.
+
+Notes:
+- This code requires Zaber devices with firmware version 7.33 or higher.
+- This code can be used to generate PVT sequence CSV files compatible with [Zaber Launcher](https://software.zaber.com/zaber-launcher/download)'s PVT Viewer App.
+
+## Dependencies
+The script uses `pipenv` to manage virtual environment and dependencies:
+
+    python3 -m pip install -U pipenv
+
+The dependencies are listed in Pipfile.
+
+## Configuration
+Edit the following constants in the script [generate_pvt_sequence.py](generate_pvt_sequence.py) to fit your setup before running the script:
+- `DATA_DIRECTORY`: the directory of the input file(s). By default
+this points to the provided 'sample_data' directory.
+- `FILENAMES`: the names of the input files to read. Instructions on
+how to structure these files can be found in the docstring of [generate_pvt_sequence.py](generate_pvt_sequence.py).
+- `TARGET_SPEED`: the target speed to use when generating velocities and times.
+- `TARGET_ACCEL`: the target aceleration to use when generating velocities and times.
+- `SHOW_PLOTS`: whether to plot the generated sequences.
+- `OUTPUT_DIRECTORY`: the directory to write the generated CSV files to. Specify this as an empty string to write to the current directory, or as None to not generate the files.
+
+## Running the Script
+To run the script:
+
+    cd src/pvt_parameter_generation
+    pipenv install
+    pipenv run python generate_pvt_sequence.py
+
+# Usage
+To quickly generate PVT data from position, position-time, or velocity-time data, follow these steps:
+1. Create a CSV file with an appropriate structure, as defined in the docstring of [generate_pvt_sequence.py](generate_pvt_sequence.py), or use one of the sample files in one of the [sample_data](sample_data/) directories:
+   - [position_data](sample_data/position_data/) contains position-only sequences,
```

Comment:
These four file links all go to the same place.

Reply:
Oops, good catch! See 9100081.